tests/test_message_broker.py

```diff
@@ -0,0 +1,105 @@
+from catkit2.catkit_bindings import LocalMemory, MessageBroker, MessageSubscriptionMode
+import numpy as np
+import pytest
+
+@pytest.fixture(scope='module')
+def broker():
+    header = LocalMemory.create(1024 * 1024 * 512)
+    block = LocalMemory.create(1024 * 1024 * 1024)
+
+    broker = MessageBroker.create(header, [block])
+    yield broker
+
+def test_message_subscription(broker):
+    topic = "test_message_subscription"
+
+    subscription_newest = broker.subscribe(topic, mode=MessageSubscriptionMode.NewestOnly)
+    subscription_sequential = broker.subscribe(topic, mode=MessageSubscriptionMode.Sequential)
+
+    for i in range(3):
+        arr = np.array([i + 10]).astype('int32')
+        message = broker.prepare_message(topic, arr.nbytes)
+        message.payload = arr
+        broker.publish_message(message)
+
+    assert subscription_newest.next_message_id == 2
+    assert subscription_sequential.next_message_id == 0
+
+    message_1 = subscription_newest.get_next_message(0.01)
+    message_2 = subscription_sequential.get_next_message(0.01)
+    message_3 = subscription_sequential.get_next_message(0.01)
+
+    assert message_1 is not None
+    assert message_2 is not None
+    assert message_3 is not None
+
+    assert message_1.payload[0] == 12
+    assert message_2.payload[0] == 10
+    assert message_3.payload[0] == 11
+
+    assert subscription_newest.try_get_next_message() is None
```

Comment:
So this means once I get a message through a subscription, that message does not exist anymore? Would make sense, just double-checking - and trying to find the correct vocabulary for this ^^